### ISSUE TITLE
Soho Modal Dialog Memory Leak

### ISSUE THREAD

**Issue by nbcp** [infor-design/enterprise-ng#636]
**Describe the bug**

**To Reproduce**
1. Go to http://localhost:4200/ids-enterprise-ng-demo/modal-dialog
2. Click Example button (Opens a modal)
3. Click Cancel / Submit (Closes the modal)
4. Generate a Heap Snapshot using Chrome DevTools' Memory tab
5. Filter objects using "modal" keyword

**Expected behavior**
It should not show references to these objects:
- Modal
- SohoModalDialogRef
- ExampleModalDialogComponent
- SohoModalDialogInjector

**Version**
- ids-enterprise-ng: latest

**Screenshots**
![image](https://user-images.githubusercontent.com/11013464/66302333-5591e100-e92b-11e9-818f-3da2e6981b13.png)

**Comment by nbcp** [comment]
So far I have came up with a partial solution which I have pushed in **soho-modal-memleak** branch. It was only able to solve 3/4 of the leaking objects (see screenshot below). Part of the changes was to remove the injector being passed into createComponent because it creates a circular dependency (injector > dialogRef > component > injector). 

What remains is to figure out how to free up **ExampleModalDialogComponent** from memory. I'm stuck in here and already ran of ideas. Need everyone's help to be able to move forward and resolve this issue.

![image](https://user-images.githubusercontent.com/11013464/66304410-ab688800-e92f-11e9-8e32-f2ca1b81540e.png)

**Comment by tmcconechy** [comment]
Yeah @nbcp i'm not 100% sure about this but i tried your branch and steps and still get all the same elements in the heap output.

I would think you should not have to do this for all variables in the object..

![Screen Shot 2019-10-07 at 11 11 01 AM](https://user-images.githubusercontent.com/814283/66324405-5870df80-e8f3-11e9-8992-1ebbc782a8ad.png)


```
import { Component, OnDestroy } from '@angular/core';

/**
 * This is an example of a simple dialog component, that can be instantiated
 * numerous times using the SohoModalDialogService.
 */
@Component({
  templateUrl: './example-modal-dialog.component.html'
})
export class ExampleModalDialogComponent implements OnDestroy {
   public model = {
    header: 'Default Header Text',
    comment: 'This task needs to be escalated to maximum priority and delivered by the end of this week.',
   };

  ngOnDestroy() {
    this.model = null;
  }
}
```

Maybe this garbage collects later or the problem is really because its not fixed in modal?

**Comment by nbcp** [comment]
Oh I forgot to mention I also had to make a little tweak in enterprise's modal.js, here's my patch: 
[modal.zip](https://github.com/infor-design/enterprise-ng/files/3700229/modal.zip)

**Comment by tmcconechy** [comment]
CC @pwpatton and @bthharper

**Comment by nbcp** [comment]
@tmcconechy sorry just ignore my previous comment, that change in enterprise wasn't really necessary.

I made a PR #639 so you guys can easily the changes in my branch. 
So far I've narrowed it down into: 
The angular component wont leak as long as it remains in its original place in DOM. However when it goes thru the angular lifecycle phases while outside its original place in DOM, the leak would still occur. See my [comment here](https://github.com/infor-design/enterprise-ng/pull/639/files#r333350174)
